src/main/java/com/clerk/backend_api/helpers/jwks/TokenVerificationException.java

```diff
@@ -0,0 +1,23 @@
+package com.clerk.backend_api.helpers.jwks;
+
+/**
+ * TokenVerificationException - Exception thrown when token verification fails.
+ */
+@SuppressWarnings("serial")
+public final class TokenVerificationException extends RuntimeException {
```

Comment:
It's probably better to have the `TokenVerificationException` as a checked exception and not `RuntimeException`, to force users to handle it.